features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewManagerScope.kt

```diff
@@ -114,7 +113,7 @@ internal class RumViewManagerScope(
         while (iterator.hasNext()) {
             val childScope = iterator.next()
             if (event is RumRawEvent.StopView) {
-                if (childScope.isActive() && (childScope as? RumViewScope)?.keyRef?.get() == event.key) {
+                if (childScope.isActive() && (childScope as? RumViewScope)?.key?.id == event.key.id) {
```

Comment:
In order to make sure that we check `id` property in the test and not the whole `key`, maybe we can modify the tests for this class (and for `RumViewScope`) to use different `RumScopeKey` for different calls (same ID, but different name/url), but it is not a blocker for me, more like something useful to add.